erpnext/controllers/accounts_controller.py

```diff
@@ -965,67 +965,197 @@ def set_advance_gain_or_loss(self):
 
 				d.exchange_gain_loss = difference
 
-	def make_exchange_gain_loss_gl_entries(self, gl_entries):
-		if self.get("doctype") in ["Purchase Invoice", "Sales Invoice"]:
-			for d in self.get("advances"):
-				if d.exchange_gain_loss:
-					is_purchase_invoice = self.get("doctype") == "Purchase Invoice"
-					party = self.supplier if is_purchase_invoice else self.customer
-					party_account = self.credit_to if is_purchase_invoice else self.debit_to
-					party_type = "Supplier" if is_purchase_invoice else "Customer"
-
-					gain_loss_account = frappe.get_cached_value(
-						"Company", self.company, "exchange_gain_loss_account"
+	def make_exchange_gain_loss_journal(self, args=None) -> None:
```

Comment:
What happens when this is called from `sales_invoice.py` or `purchase_invoice.py`?  I can only see handling for Journal Entry and Payment Entry

Reply:
Existing design of linking payments or journals to other Doctypes requires modiyfing the said payments or journals.
References table for Payment Entry and Accounts table in Journals.

The Exchange Gain/Loss booking logic relies only this existing design. Wether payment is made as an advance or normal payment, whether linked during invoice creation or reconciled later, the References table will be updated. Same is true for Journals.

We are using this existing design to make Exchange Gain Or Loss only from Payment Entries or Journal Entries.